pinot-core/src/main/java/com/linkedin/pinot/core/indexsegment/generator/SegmentPartitionConfig.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.indexsegment.generator;
+
+import com.linkedin.pinot.core.data.partition.PartitionFunction;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.apache.commons.lang.math.IntRange;
+import org.codehaus.jackson.annotate.JsonIgnoreProperties;
+
+
+@SuppressWarnings("unused") // Suppress incorrect warning, as methods are used for json ser/de.
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class SegmentPartitionConfig {
+  Map<String, ColumnPartitionConfig> _columnPartitionMap;
```

Comment:
Made private. Cannot be final because empty constructor required for json de-serialization